packages/flutter_tools/lib/src/build_system/targets/android.dart

```diff
@@ -274,15 +300,24 @@ class AndroidAotBundle extends Target {
   String get name => 'android_aot_bundle_${getNameForBuildMode(dependency.buildMode)}_'
     '${getNameForTargetPlatform(dependency.targetPlatform)}';
 
+  TargetPlatform get targetPlatform => dependency.targetPlatform;
+
+  /// The selected build mode.
+  ///
+  /// This is restricted to [BuildMode.profile] or [BuildMode.release].
+  BuildMode get buildMode => dependency.buildMode;
+
   @override
   List<Source> get inputs => <Source>[
-   Source.pattern('{BUILD_DIR}/$_androidAbiName/app.so'),
+    Source.pattern('{BUILD_DIR}/$_androidAbiName/app.so'),
+    Source.pattern('{BUILD_DIR}/$_androidAbiName/manifest.json'),
```

Comment:
The manifest doesn't get produced if its not a split AOT right?

Reply:
Correct. I'll change this to track the manifest in the depfile in case the app does not include any loading units.